screens/gallery/GalleryScreen.js

```diff
@@ -0,0 +1,289 @@
+};
+
+class GalleryScreen extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedImages: [],
+            sortedData: []
+        };
+    }
+
+    componentDidMount() {
+        this.splitIntoRows(this.props.geotaggedImages);
+    }
+
+    async splitIntoRows(images) {
+        let temp = {};
+        images.map(image => {
+            const timestampOfImage = image.timestamp * 1000;
+            const placeInTimeOfImage = placeInTime(timestampOfImage);
+
+            if (temp[placeInTimeOfImage] === undefined) {
+                temp[placeInTimeOfImage] = [];
+            }
+            temp[placeInTimeOfImage].push(image);
+        });
+
+        let final = [];
+        let order = ['today', 'week', 'month'];
+        let allTimeTags = Object.keys(temp).map(prop => {
+            if (Number.isInteger(parseInt(prop))) {
+                return parseInt(prop);
+            }
+
+            return prop;
+        });
+        let allMonths = allTimeTags.filter(
+            prop => Number.isInteger(prop) && prop < 12
+        );
+        allMonths = _.reverse(_.sortBy(allMonths));
+        let allYears = allTimeTags.filter(
+            prop => Number.isInteger(prop) && !allMonths.includes(prop)
+        );
+        allYears = _.reverse(_.sortBy(allYears));
+
+        order = _.concat(order, allMonths, allYears);
+
+        for (let prop of order) {
+            if (temp[prop]) {
+                let newObj = { title: prop, data: temp[prop] };
+                final.push(newObj);
+            }
+        }
+        this.setState({ sortedData: final });
+    }
+
+    /**
+     * fn that is called when "done" is pressed
+     * sorts the array based on id
+     * call action photosFromGallery to save selected images to state
+     *
+     * saves the selected array of images to Async store
+     */
+    async handleDoneClick() {
+        const sortedArray = await this.state.selectedImages.sort(
+            (a, b) => a.id - b.id
+        );
+
+        this.props.photosFromGallery(sortedArray);
+
+        AsyncStorage.setItem(
+            'openlittermap-gallery',
+            JSON.stringify(this.state.selectedImages)
+        ).then(_ => {
+            return true;
+        });
+    }
+
+    /**
+     * fn to select and deselect image on tap
+     *
+     * @param  item - The image object
+     */
+
+    selectImage(item) {
+        const selectedArray = this.state.selectedImages;
+        // check if item /image is already selected
+        const index = selectedArray.indexOf(item);
+        if (index !== -1) {
+            this.setState({
+                selectedImages: this.state.selectedImages.filter(
+                    (_, i) => i !== index
+                )
+            });
+        }
+
+        if (index === -1) {
+            this.setState(prevState => {
+                return { selectedImages: [...prevState.selectedImages, item] };
+            });
+        }
+    }
```

Comment:
Instead of having 2 arrays, it might be simpler if we added `selected: bool` prop to each image?

```
selectImage (id)
{
     let images = [...state.images];
     
     let img = images.findById(id);
     
     img.selected = ! image.selected;
     
     state.images = images;
}
```

Reply:
Few things
- I am not saving the whole geotagged images array in state initially, I am only saving selected images to local state.
- I don't want to add a selected boolean to the global geotagged array item. because there is already one that was already there which is used for selecting for DELETE I think.. and I dont want to add more booleans to make it more confusing
- and about the suggested code. I dont understand it fully .... I dont think there is anything like findById that I know of. and images is an array of objects and id is key of that so even indexOf will not work here. So eventually the suggested code will also be similar to mine I think... Please let me know if I am wrong

Reply:
`selected` could be re-used. `findById` was psuedo code for `array.find(item => item.id === id);`.  We can accept 2 arrays for now and see how it goes.